crates/ra_syntax/src/parsing/reparsing.rs

```diff
@@ -162,20 +162,27 @@ fn is_balanced(tokens: &[Token]) -> bool {
 fn merge_errors(
     old_errors: Vec<SyntaxError>,
     new_errors: Vec<SyntaxError>,
-    old_range: TextRange,
+    range_before_reparse: TextRange,
     edit: &AtomTextEdit,
 ) -> Vec<SyntaxError> {
     let mut res = Vec::new();
-    for e in old_errors {
-        if e.offset() <= old_range.start() {
-            res.push(e)
-        } else if e.offset() >= old_range.end() {
-            res.push(e.add_offset(TextUnit::of_str(&edit.insert), edit.delete.len()));
+
+    for old_err in old_errors {
+        let old_err_range = *old_err.range();
+        // FIXME: make sure that .start() was here previously by a mistake
```

Comment:
This fixme will be removed before merging this PR

Reply:
It should be TODO then

Reply:
Yes, I just temporarily marked it FIXME to let it pass `no todo` tests for a while.